src/app/courses/pages/courses-page/courses-page.component.html

```diff
@@ -1 +1,13 @@
-<p>courses-page works!</p>
+<div class="courses-container">
+  <app-search-course></app-search-course>
+  <div class="add-course">
+    <button class="btn">Add Course</button>
+  </div>
+  <ng-container *ngFor="let course of courses">
```

Comment:
It would be better to create a separate component CoursesList.
For example, if in the future you will need a courses list on another page, now you have to copy and paste all related to courses list logic from the courses page to another page.

If you have a separate CoursesList component, you just need to reuse the CoursesList component on another page.

Reply:
fixed thanks

Reply:
You forgot about trackBy.